include/seqan3/core/algorithm/all.hpp

```diff
@@ -38,13 +38,68 @@
  */
 
 #pragma once
-//!\cond DEV
+
 /*!\defgroup algorithm Algorithm
  * \ingroup core
  * \brief Contains core functionality used to configure algorithms.
- * \todo Write detailed description about usage of this module.
+ *
+ * \details
+ *
+ * In SeqAn there are many, e.g. alignment or search interfaces, that can be configured through
+ * many different settings and policies that alter the execution of the respective algorithm.
+ * These configurations can be orthogonal or might be mutually exclusive and can make interfaces very difficult to use.
+ * This module provides a basic system to manage the configurations of algorithms using a unified interface.
+ *
+ * ### Usage
+ *
+ * The basis of any algorithm configuration are configuration elements. These are objects that handle a specific
+ * setting and must satisfy the seqan3::detail::config_element_concept. The following snippet demonstrates a
+ * basic setup for such configuration elements.
+ *
+ * \snippet test/snippet/core/algorithm/configuration.cpp configuration_setup
+ *
+ * Here, two classes with the name `bar` and `foo` are created. They can be normal or template classes and must
+ * inherit from seqan3::pipeable_config_element and contain a member with the name `value` to satisfy the respective
+ * concept. The separate `value` member is used for a proper encapsulation from the actual setting parameter.
+ * For example the alignment algorithms require a scoring scheme, but the scoring scheme itself should not be pipeable
+ * with other settings.
+ *
+ * In addition an enum type was defined that will be used later to allow for compatibility checks when combining
+ * settings in a configuration object. This enum assigns every configuration element a unique id.
+ * To provide compatibility checks the `seqan3::detail::compatibility_table` must be overloaded for the specific
+ * algorithm configuration.
+ *
+ * \snippet test/snippet/core/algorithm/configuration.cpp compatibility
+ *
+ * The type for the configuration element ids is used to overload the bool table. In the example above, both
+ * elements can be combined with each other but not with themselves, to avoid inconsistent settings.
```

Comment:
Does the table always need to be symmetric?

Reply:
yes. Why would you allow local with global but not global with local. The order should not matter, becuase this would be confusing for the users and the error detection very cumbersome.